libsplinter/src/auth/oauth/rest_api/actix/login.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The `GET /oauth/login` endpoint for getting the authorization request URL for the provider.
+
+use actix_web::{http::header::LOCATION, HttpResponse};
+use futures::future::IntoFuture;
+
+use crate::auth::oauth::OAuthClient;
+use crate::protocol;
+use crate::rest_api::{ErrorResponse, Method, ProtocolVersionRangeGuard, Resource};
+
+pub fn make_login_route(client: OAuthClient) -> Resource {
```

Comment:
Calling this route /login is a bit confusing in that its pretty common to POST to a /login route for username/password auth, where after the operation you are logged in.

Reply:
Would `/oauth/authorize` be better?

Reply:
Almost feels more like retrieving config -- /oauth/config or something. Maybe we work it out in a subsequent PR.